src/OpenTelemetry.AutoInstrumentation.NLogTarget/OpenTelemetryTarget.cs

```diff
@@ -0,0 +1,316 @@
+    private static readonly bool SupportsTypedLayouts = CheckTypedLayoutSupport();
+
+    private static LoggerProvider? _loggerProvider;
+    private static Func<string?, object?>? _getLoggerFactory;
+    
+    // Typed layouts for NLog 5.3.4+ optimization
+    private Layout<ActivityTraceId?>? _typedTraceIdLayout;
+    private Layout<ActivitySpanId?>? _typedSpanIdLayout;
+
+    public OpenTelemetryTarget()
+    {
+        Layout = "${message}";
+        // Use property setters to properly initialize both string and typed layouts
+        TraceIdLayout = "${activity:property=TraceId}";
+        SpanIdLayout = "${activity:property=SpanId}";
+    }
+
+    [RequiredParameter]
+    public Layout? Endpoint { get; set; }
+
+    public Layout? Headers { get; set; }
+
+    public bool UseHttp { get; set; } = true;
+
+    public Layout? ServiceName { get; set; }
+
+    [ArrayParameter(typeof(TargetPropertyWithContext), "resource")]
+    public IList<TargetPropertyWithContext> Resources { get; } = new List<TargetPropertyWithContext>();
+
+    public bool IncludeFormattedMessage { get; set; } = true;
+
+    public bool IncludeEventParameters { get; set; } = true;
+
+    private Layout? _traceIdLayout;
+    private Layout? _spanIdLayout;
+    
+    public Layout? TraceIdLayout 
+    { 
+        get => _traceIdLayout;
+        set
+        {
+            _traceIdLayout = value;
+            // Update typed layout if supported
+            if (SupportsTypedLayouts && value != null)
+            {
+                _typedTraceIdLayout = value.Text;
+            }
+        }
+    }
+
+    public Layout? SpanIdLayout 
+    { 
+        get => _spanIdLayout;
+        set
+        {
+            _spanIdLayout = value;
+            // Update typed layout if supported
+            if (SupportsTypedLayouts && value != null)
+            {
+                _typedSpanIdLayout = value.Text;
+            }
+        }
+    }
+
+    public int ScheduledDelayMilliseconds { get; set; } = 5000;
+
+    public int MaxQueueSize { get; set; } = 2048;
+
+    public int MaxExportBatchSize { get; set; } = 512;
+
+    protected override void InitializeTarget()
+    {
+        base.InitializeTarget();
+
+        if (_loggerProvider != null)
+        {
+            return;
+        }
+
+        var createLoggerProviderBuilderMethod = typeof(Sdk).GetMethod("CreateLoggerProviderBuilder", System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic)!;
+        var loggerProviderBuilder = (LoggerProviderBuilder)createLoggerProviderBuilderMethod.Invoke(null, null)!;
+
+        loggerProviderBuilder = loggerProviderBuilder
+            .SetResourceBuilder(ResourceConfigurator.CreateResourceBuilder(Instrumentation.GeneralSettings.Value.EnabledResourceDetectors));
+
+        loggerProviderBuilder = loggerProviderBuilder.AddOtlpExporter(options =>
+        {
+            var endpoint = RenderLogEvent(Endpoint, LogEventInfo.CreateNullEvent());
```

Comment:
With my recommendation that we do not have programmatic configuration, we can rely solely on OTEL environment variables for the OTLP exporter settings, so I think we can remove the calls to `RenderLogEvent` here